datajunction-server/datajunction_server/api/helpers.py

```diff
@@ -739,7 +739,7 @@ async def build_sql_for_multiple_metrics(  # pylint: disable=too-many-arguments,
                 # It must be either a Source node or some type of Materialized node.
                 # Even if there is no availability state for it right now, this qery may be built
                 # for future execution, so we don't need to check for availability state here.
-                if tbl.dj_node
+                if tbl.dj_node and tbl.dj_node.type == NodeType.SOURCE
```

Comment:
I see what you are saying... but this seems to differ from the comment lines 740-741. No?

Reply:
Hmm, I don't think we should include nodes with no availability state in this list of upstream tables though. I added a check to include nodes with a materialization. 

I guess the real question is the purpose of this list -- we're currently using it as a way of determining the list of upstream physical tables that this generated query references, but perhaps we shouldn't use it that way?